### ISSUE TITLE
Seg Fault When Creating Sprite

### ISSUE THREAD

**Issue by Ktoks** [bevyengine/bevy#2661]
## Bevy version = 0.5

Commit Hash = 9d453530fac9202691c797184ad0b220b2ea37b3

## Operating system & version

Windows 10 build 21390.co_release.210521-1658 , and Ubuntu 20.04 (on WSL2)

## What you did

The following code:
```rs
use bevy::prelude::*;

const PLAYER_SPRITE: &str = "player_a_01.png";

// entity, component, system, resource

fn main() {
    App::build()
        .insert_resource(ClearColor(Color::rgb(0.04, 0.04, 0.04)))
        .insert_resource(WindowDescriptor {
            title: "Rust Invaders!".to_string(),
            width: 598.0,
            height: 676.0,
            ..Default::default()
        })
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup.system())
        .run();
}

fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<ColorMaterial>>,
    mut windows: ResMut<Windows>,
) {
    // camera
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());

    // position window
    let window = windows.get_primary_mut().unwrap();
    window.set_position(IVec2::new(1920, 1080));

    // spawn sprite
    commands.spawn_bundle(SpriteBundle {
        material: materials.add(asset_server.load(PLAYER_SPRITE).into()),
        ..Default::default()
    });
}

```

## What you expected to happen

Create a sprite- display it on the screen in a window.

## What actually happened

Segmentation fault immediately

## Additional information

Bevy examples now work on windows Machine.

**Comment by kacystocks** [comment]
If you want to see what I was attempting to build- I was following the tutorial at the following link:

[https://youtu.be/Yb3vInxzKGE](https://youtu.be/Yb3vInxzKGE)

**Comment by mockersf** [comment]
I don't think anything in that code should segfault... Do you have more details?

Could you retry without the `position window` block which is the only somewhat unusual code that I see?
```rust
    // position window
    let window = windows.get_primary_mut().unwrap();
    window.set_position(IVec2::new(1920, 1080));
```

**Comment by kacystocks** [comment]
Sure can, I'll give it a shot. Sorry for the delayed response I've been sick.

**Comment by kacystocks** [comment]
So it works on my windows machine, but still segfaults on WSL Ubuntu.

**Comment by kacystocks** [comment]
After further testing- it causes a segfault in the line:
```rust
.add_plugins(DefaultPlugins)
```

**Comment by kacystocks** [comment]
I don't know if this is a useful error, I commented out the .add_plugins... line from above:
thread 'main' panicked at 'Requested resource does not exist: bevy_asset::asset_server::AssetServer', /home/kacy/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.5.0/src/system/system_param.rs:244:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by shinaka** [comment]
This is actually because alsa can't find a sound device at the default '0' in WSL2. I'm not entirely sure how to fix this though, as wslg has pulseaudio setup and other apps have no issue playing audio.

**Comment by AxiomaticSemantics** [comment]
This issue should probably be renamed to: `Pulseaudio on WSL2 causes crash on startup due to fine finding audio device`

**Comment by SolarLiner** [comment]
Fixed by #2269
